index.bs

```diff
@@ -603,16 +603,66 @@ ReadableStream(<var>underlyingSource</var> = {}, { <var>size</var>, <var>highWat
+       * While WritableStreamDefaultWriterGetDesiredSize(_writer_) is ≤ *0* or is *null*, the user agent must not read
+         from _reader_.
+       * If _reader_ is a <a>BYOB reader</a>, WritableStreamDefaultWriterGetDesiredSize(_writer_) should be used to
+         determine the size of the chunks read from _reader_.
+       * Otherwise, WritableStreamDefaultWriterGetDesiredSize(_writer_) may be used to determine the flow rate
+         heuristically, e.g. by delaying reads while it is judged to be "low" compared to the size of chunks that have
+         been typically read.
+     * Errors must be propagated forward:
+       * If *this*.[[state]] is or becomes `"errored"` and _preventAbort_ is *false*, perform !
+         WritableStreamAbort(_dest_, *this*.[[storedError]]). If the resulting promise fulfills, perform error cleanup
+         with *this*.[[storedError]]. Otherwise, perform error cleanup with the resulting promise rejection.
+       * If *this*.[[state]] is or becomes `"errored"` and _preventAbort_ is *true*, perform error cleanup with
+         *this*.[[storedError]].
+     * Errors must be propagated backward:
+       * If _dest_.[[state]] is or becomes `"errored"` and _preventCancel_ is *false*, perform !
+         ReadableStreamCancel(*this*, _dest_.[[storedError]]). If the resulting promise fulfills, perform error cleanup
+         with _dest_.[[storedError]]. Otherwise, perform error cleanup with the resulting promise rejection.
+       * If _dest_.[[state]] is or becomes `"errored"` and _preventCancel_ is *true*, perform error cleanup with
+         _dest_.[[storedError]].
+     * Closing must be propagated forward:
+       * If *this*.[[state]] is or becomes `"closed"` and _preventClose_ is *false*, perform !
+         WritableStreamDefaultWriterClose(_writer_). If the resulting promise fulfills, perform successful cleanup.
+         Otherwise, perform error cleanup with the resulting promise rejection.
+       * If *this*.[[state]] is or becomes `"closed"` and _preventClose_ is *true*, perform successful cleanup.
+     * Closing must be propagated backward:
+       * If _dest_.[[state]] is `"closing"` or `"closed"` and _preventCancel_ is *false*,
+         1. Let _destClosed_ be a new *TypeError*.
+         1. Perform ! ReadableStreamCancel(*this*, _destClosed_).
+         1. If the resulting promise fulfills, perform error cleanup with _destClosed_. Otherwise, perform error cleanup
+            with the resulting promise rejection.
+       * If _dest_.[[state]] is `"closing"` or `"closed"` and _preventCancel_ is *true*, perform error cleanup with a
+         new *TypeError*.
+     * <i>Cleanup</i>: if any of the above requirements ask to perform cleanup, then:
+       * Wait until any outstanding reads or writes finish (i.e. the corresponding promises settle). If that causes one
+         of the above error conditions to be true, jump back to those requirements instead of proceeding with cleanup.
```

Comment:
Depending on resolution of #476, we might need to mention the error situation where write() promise rejects but the destination writable stream doesn't get errored.